ThunkLibs/GuestLibs/CMakeLists.txt

```diff
@@ -173,3 +181,14 @@ generate(libdrm ${CMAKE_CURRENT_SOURCE_DIR}/../libdrm/libdrm_interface.cpp thunk
 target_include_directories(libdrm-guest-deps INTERFACE /usr/include/drm/)
 target_include_directories(libdrm-guest-deps INTERFACE /usr/include/libdrm/)
 add_guest_lib(drm)
+
+generate(libVDSO ${CMAKE_CURRENT_SOURCE_DIR}/../libVDSO/libVDSO_interface.cpp thunks function_packs function_packs_public)
+add_guest_lib(VDSO)
+# Can't use a stack protector because otherwise cross-compiling fails
+# Not necessary anyway because it only trampolines
+target_compile_options(VDSO-guest PRIVATE "-fno-stack-protector")
+target_link_options(VDSO-guest PRIVATE "-T" "${CMAKE_CURRENT_SOURCE_DIR}/../libVDSO/libVDSO_Guest.lds" "-nostdlib" "-Wl,-soname,linux-vdso.so.1"
+  "-Wl,--no-undefined" "-Wl,-z,max-page-size=4096" "-Wl,--hash-style=both")
+
+# Need to override SONAME in linker options
```

Comment:
Could you elaborate on why, or is that something that's obvious when you're more familiar with VDSO?

Reply:
Likely the soname also needs to be overridden for all guest thunk libraries to match what the guest libraries would have.

Myself I don't exactly know how soname (DT_SONAME entry) affects things, apart that it is a "canonical name for the library"

Reply:
Every thunk guest library now sets its SONAME directly with a comment in the cmake file as to why.